docs/programming-guide.md

```diff
@@ -743,6 +855,30 @@ def doStuff(self, rdd):
 
 </div>
 
+<div data-lang="r"  markdown="1">
+
+Spark's API relies heavily on passing functions in the driver program to run on the cluster.
+There are three recommended ways to do this:
+
+* [Anonymous functions](http://adv-r.had.co.nz/Functional-programming.html#anonymous-functions),
+  for simple functions that can be written as an anonymous function.
+* Top-level functions in a module.
+
+For example, to pass a longer function than can be supported using a `lambda`, consider
```

Comment:
Is this a carryover from the PySpark docs? Probably better not to use `lambda` in the R version.